plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -425,6 +430,10 @@ func (s *SQLServer) Init() error {
 		s.Log.Warn("Warning: Server list is empty.")
 	}
 
+	if s.Timeout <= 0 {
+		s.Timeout = config.Duration(5 * time.Second)
+	}
+
```

Comment:
I'm generally good with adding an option like this, but I am not good with setting a default that changes the behavior.

A setting of "0s" should mean no timeout, which should also be the default and stated as such in the config.

Reply:
@powersj i do agree with you 0 should mean no timeout , I took this as an example from the sql plugin (https://github.com/influxdata/telegraf/blob/176a416f640dec0483991326f8b1c882246e1d3a/plugins/inputs/sql/sql.go#L232) . **The behaviour of telegraf should be consistent**

Reply:
I also agree that consistency should be something we strive for. In this case, the consistency that matters more is not between plugins and instead that the experience using the sqlserver plugin continues to act in a manner they are familiar with.

Surprising users suddenly with a change in behavior, which could suddenly stop collecting data, cause failures, and requires them to figure out how long of a timeout they need is less than user friendly.

Reply:
Well explained , i will make necessary changes.

Reply:
@powersj  code review comment has been resolved , please review again